SU2_CFD/include/solvers/CFVMFlowSolverBase.inl

```diff
@@ -1656,6 +1658,8 @@ void CFVMFlowSolverBase<V, FlowRegime>::SetResidual_DualTime(CGeometry *geometry
     }
     END_SU2_OMP_FOR
 
+    AD::EndNoSharedReading();
+
```

Comment:
Some of these shared reading optimizations depend on `TimeStep` being passive. Does this always hold true?

Reply:
It's a fair-enough assumption, I added some more over the Primitive loops and removed some over smaller loops where the performance benefit might not justify the increased maintenance.